test/OrderBy.spec.js

```diff
@@ -0,0 +1,29 @@
+const expect = require('chai').expect;
+require('..')();
+describe('Array#prototype#Reverse', function () {
+
+    it('Should return sorted based on provided key for all same types', function () {
+        const inputSameType = [{ x: 3, y: 9 },{ x: 2, y: 4 }];
+        const expectedSameType = [{ x: 2, y: 4 }, { x: 3, y: 9 }];
+        const actualSameType = inputSameType.OrderBy(x);
+        expect(actualSameType).to.eql(expectedSameType);
+    });
+    it('Should return sorted based on provided key for mixed types', function () {
+        const inputMixedType = [{ x: 2, y: 'b' }, { x: 3, y: 'a' }];
+        const expectedMixedType = [{ x: 3, y: 'a' }, { x: 2, y: 'b' }];
+        const actualMixedType = inputMixedType.OrderBy(y);
```

Comment:
`y` is not defined. `keySelector` must be a function.

Reply:
Fixed